packages/server/src/db/index.ts

```diff
@@ -146,7 +146,7 @@ export class Database {
       .selectFrom('repo_root')
       .selectAll()
       .where('did', '=', did)
-    if (forUpdate && this.dialect !== 'sqlite') {
+    if (forUpdate && this.dialect !== 'sqlite' && this.isTransaction) {
```

Comment:
This seems reasonable to me, though in sqlite this could be run outside of a transaction in which case no real locking would happen.  If that matters to us, could do something like this:
```js
if (forUpdate) {
  this.assertTransaction()
  if (this.dialect !== 'sqlite') {
    builder = builder.forUpdate()
  }
}
```